storage/pin/pin_test.go

```diff
@@ -0,0 +1,328 @@
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package pin
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"io/ioutil"
+	"mime"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/ethersphere/swarm/api"
+	"github.com/ethersphere/swarm/chunk"
+	"github.com/ethersphere/swarm/storage"
+	"github.com/ethersphere/swarm/storage/feed"
+	"github.com/ethersphere/swarm/storage/localstore"
+	"github.com/ethersphere/swarm/testutil"
+)
+
+// Pin a RAW file and unpin it multiple times
+func TestPinRawUpload(t *testing.T) {
+
+	p, f := getPinApiAndFileStore(t)
+	data := testutil.RandomBytes(1, 10000)
+	hash := uploadFile(t, f, data, false)
+
+	// test pin and unpin
+	pinUnpinAndFailIfError(t, p, hex.EncodeToString(hash), 5, true)
+
+}
+
+// Pin a encrypted RAW file and unpin it multiple times
+func TestPinRawUploadEncrypted(t *testing.T) {
+
+	p, f := getPinApiAndFileStore(t)
+	data := testutil.RandomBytes(1, 10000)
+	hash := uploadFile(t, f, data, true)
+
+	// test pin and unpin
+	pinUnpinAndFailIfError(t, p, hex.EncodeToString(hash), 5, true)
+
+}
+
+// Pin a simple collection and unpin it multiple times
+func TestPinCollectionUpload(t *testing.T) {
+
+	p, f := getPinApiAndFileStore(t)
+	hash := uploadCollection(t, p, f, false)
+
+	// test pin and unpin
+	pinUnpinAndFailIfError(t, p, hex.EncodeToString(hash), 1, false)
+
+}
+
+// Pin a encrypted simple collection and unpin it multiple times
+func TestPinCollectionUploadEncrypted(t *testing.T) {
+
+	p, f := getPinApiAndFileStore(t)
+	hash := uploadCollection(t, p, f, true)
+
+	// test pin and unpin
+	pinUnpinAndFailIfError(t, p, hex.EncodeToString(hash), 5, false)
+
+}
+
+// Test if the uploaded collection shows up in the ListPinFiles command
+func TestListPinInfo(t *testing.T) {
+
+	p, f := getPinApiAndFileStore(t)
+	hash := uploadCollection(t, p, f, false)
+
+	// Pin the hash
+	err := p.PinFiles(hex.EncodeToString(hash), false, "")
+	if err != nil {
+		t.Fatalf("Could not pin " + err.Error())
+	}
+
+	// Gte the list of file pinned
+	pinInfo := p.ListPinFiles()
+
+	// Check if the uploaded collection is in the list files
+	if _, ok := pinInfo[hex.EncodeToString(hash)]; !ok {
+		t.Fatalf("uploaded collection not pinned")
+	}
+
+}
+
+func getPinApiAndFileStore(t *testing.T) (*PinAPI, *storage.FileStore) {
+
+	t.Helper()
+
+	swarmDir, err := ioutil.TempDir("", "swarm-storage-test")
```

Comment:
`defer os.RemoveAll(swarmDir)`
please go through all of the tests and helper functions and make sure this is done

Reply:
Done